goldens/public-api/angular_devkit/architect/index.md

```diff
@@ -49,8 +49,8 @@ export interface BuilderContext {
 }
 
 // @public
-export interface BuilderHandlerFn<A> {
-    (input: A, context: BuilderContext): BuilderOutputLike;
+export interface BuilderHandlerFn<Options, InfrastructureSettings = {}> {
```

Comment:
While, I understand why we need `infrastructureSettings` since we want to override certain "unknown" options when running the builder.

Example
```ts
  const browserTargetRun = await context.scheduleTarget(browserTarget);
```

I think there are a couple of things that we need to understand and agree on here. 
- The new test builder will be coupled with the implementation of the browser builder. In this case the CLI is opinionated that users **cannot** use any other builder if you want to use the new test runner, or if they want, the new builder would need to support the same options that the in-build Angular CLI builder supports. 
- This will increase improbabilities which custom builders and also make it almost impossible to use the new test builder with another browser builder, let's say NX.

IMO, I don't think we should couple this, and currently the test builder is not coupled with the browser builder API. Example, currently users can use esbuild/rspack/webpack to build their application and use webpack to run their tests through karma.

I don't want to block this PR, but I feel like this is a discussion that we need to have, as if we don't want to highly couple ourselves our browser builder, which would make it harder for users to use 3P builders etc.

If we are going with pre-bundling, I do believe that we should not invoke the builder through architect, but rather invoke the internal esbuild method, and the reason for this is as by using the architect builder directly we are actually increasing complexity, coupling the browser and test builder for just re-using a couple of options.  

**Option to re-use**
- `fileReplacements`- which I am iffy if we should add this in the test builder.
- `inlineStyleLanguage` - if we want to support component styles
- `stylePreprocessorOptions`- if we want to support component styles
- `externalDependencies` - not sure if this is needed for testing purposes, as this will likely cause problems.

**Options to override**
- `sourceMap` - this options also depends on the code-coverage value as when this is set to true sourcemaps also need to be true
- `watch` - this is based on if the test runner is running in watch mode.
- `aot` - tests always in JIT
- `index` - not needed
- `dist`
- `main` - overridden to entrypoints.
- `tsConfig` - tsconfig.spec.ts
- `progress` - based on the test builder config
- `serviceWorker` - always disabled
- `scripts` - not needed for tests
- `styles` - not needed for tests
- `webWorkerTsConfig` - not supported for tests
- `budgets` - disabled for tests
- `assets` - not needed for tests.
- `polyfills` - are different for test

Eventually more and more options will need to be overridden.